src/phases/turn-start-phase.ts

```diff
@@ -42,21 +43,19 @@ export class TurnStartPhase extends FieldPhase {
       orderedTargets = Utils.randSeedShuffle(orderedTargets);
     }, this.scene.currentBattle.turn, this.scene.waveSeed);
 
+    // Next, a check for Trick Room is applied to determine sort order.
+    const speedReversed = new Utils.BooleanHolder(false);
+    this.scene.arena.applyTags(TrickRoomTag, speedReversed);
+    applyChallenges(this.scene.gameMode, ChallengeType.TRICK_ROOM, speedReversed);
+
+    // Adjust the sort function based on whether Trick Room is active.
     orderedTargets.sort((a: Pokemon, b: Pokemon) => {
       const aSpeed = a?.getEffectiveStat(Stat.SPD) || 0;
       const bSpeed = b?.getEffectiveStat(Stat.SPD) || 0;
```

Comment:
```suggestion
      const aSpeed = a?.getEffectiveStat(Stat.SPD) ?? 0;
      const bSpeed = b?.getEffectiveStat(Stat.SPD) ?? 0;
```
Might as well change these while you're here.